src/main/check.cpp

```diff
@@ -32,6 +32,44 @@ struct at_scope_exit
     ~at_scope_exit() { on_exit(); }
 };
 
+static const std::map<std::string, bpf_conformance_groups_t> _conformance_groups = {
+    {"atomic32", bpf_conformance_groups_t::atomic32},
+    {"atomic64", bpf_conformance_groups_t::atomic64},
+    {"base32", bpf_conformance_groups_t::base32},
+    {"base64", bpf_conformance_groups_t::base64},
+    {"callx", bpf_conformance_groups_t::callx},
+    {"divmul32", bpf_conformance_groups_t::divmul32},
+    {"divmul64", bpf_conformance_groups_t::divmul64},
+    {"packet", bpf_conformance_groups_t::packet}};
+
+static std::optional<bpf_conformance_groups_t> _get_conformance_group_by_name(std::string group) {
+    if (!_conformance_groups.contains(group)) {
+        return {};
+    }
+    return _conformance_groups.find(group)->second;
+}
+
+static std::string _get_conformance_group_names() {
+    std::string result;
+    for (const auto& entry : _conformance_groups) {
+        if (!result.empty()) {
+            result += ", ";
+        }
+        result += entry.first;
+    }
+    return result;
+}
+
+// Given a string containing comma-separated tokens, split them into a list of strings.
+static std::vector<std::string> get_string_vector(std::string list) {
+    std::vector<std::string> string_vector;
+    std::string element;
+    for (std::istringstream iss(list); std::getline(iss, element, ',');) {
+        string_vector.push_back(element);
+    }
```

Comment:
```suggestion
    boost::split(string_vector, str, boost::is_any_of(","));
```

And include
```c++
#include <boost/algorithm/string.hpp>
```

(Though it's probably time we use `string_view`)

Reply:
Updated as suggested